Remove chuck norris jokes, replace with JSON placeholder todos (#32661)
dotnet/docs: docs/core/extensions/httpclient-factory.md

## The `IHttpClientFactory` type

> All of the sample source code in this article relies on the [`Microsoft.Extensions.Http`](https://www.nuget.org/packages/microsoft.extensions.http) NuGet package. Additionally, [{JSON} Placeholder](https://jsonplaceholder.typicode.com/) free API is used to make HTTP `GET` requests for user todo objects.

Reviewer: ```suggestion
All of the sample source code in this article relies on the [`Microsoft.Extensions.Http`](https://www.nuget.org/packages/microsoft.extensions.http) NuGet package. Additionally, HTTP `GET` requests are made to the free [{JSON} Placeholder](https://jsonplaceholder.typicode.com/) API to get user `Todo` objects.
```

Reviewer: I don't think "todo" would translate well.

A typed client accepts an `HttpClient` parameter in its constructor:

:::code source="snippets/http/typed/TodoService.cs" highlight="9,13,15,23-25":::

In the preceding code:

- The configuration is set when the typed client is added to the service collection.
- The `HttpClient` is assigned as a class-scoped variable (field), and used with exposed APIs.

> API-specific methods can be created that expose `HttpClient` functionality. For example, the `GetUserTodosAsync` method encapsulates code to retrieve user specific todo objects.

Reviewer: ```suggestion
API-specific methods can be created that expose `HttpClient` functionality. For example, the `GetUserTodosAsync` method encapsulates code to retrieve user-specific `Todo` objects.
```
